src/oncall/auth/modules/ldap_example.py

```diff
@@ -1,33 +1,67 @@
 # Copyright (c) LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
 # See LICENSE in the project root for license information.
 import ldap
+import os
+import logging
 
+logger = logging.getLogger(__name__)
 
 class Authenticator:
     def __init__(self, config):
         if config.get('debug'):
             self.authenticate = self.debug_auth
             return
+        self.authenticate = self.ldap_auth
+
+        if 'ldap_cert_path' in config:
+            self.cert_path = config['ldap_cert_path']
+            if not os.access(self.cert_path, os.R_OK):
+                logger.error("Failed to read certificate")
+                return False
+
+        if 'ldap_bind_user' in config:
+            self.bind_user = config['ldap_bind_user']
+            self.bind_password = config['ldap_bind_password']
+            self.search_filter = config['ldap_search_filter']
+
         self.ldap_url = config['ldap_url']
-        self.cert_path = config['ldap_cert_path']
         self.user_suffix = config['ldap_user_suffix']
-        self.authenticate = self.ldap_auth
+        self.base_dn = config['ldap_base_dn']
+
 
     def ldap_auth(self, username, password):
-        ldap.set_option(ldap.OPT_X_TLS_CACERTFILE, self.cert_path)
+        bind_user = username + self.user_suffix
+        bind_password = password
+
+        try:
+            ldap.set_option(ldap.OPT_X_TLS_CACERTFILE, self.cert_path)
+            # override bind user and password if they have been specified in the config
+            bind_user = self.bind_user
+            bind_password = self.bind_password
+        except AttributeError:
+            pass
+
         connection = ldap.initialize(self.ldap_url)
         connection.set_option(ldap.OPT_REFERRALS, 0)
 
         try:
             if password:
-                connection.simple_bind_s(username + self.user_suffix, password)
+                connection.simple_bind_s(bind_user, bind_password)
```

Comment:
This has two different actions depending on whether self.bind_user exists. If it doesn't exist, it acts as the user credentials check. Otherwise, it binds the connection using the config's user/password. I suggest refactoring to move the `if self.bind_user:` block above this line, having that block bind the config user. Then, this line will have the consistent purpose of checking the user's credentials.